crates/engine/tree/src/tree/payload_processor/prewarm.rs

```diff
@@ -253,35 +213,47 @@ where
         Some((evm, evm_config, metrics))
     }
 
-    /// Transacts the transaction and returns the state outcome.
+    /// Transacts the vec of transactions and returns the state outcome.
     ///
-    /// Returns `None` if executing the transaction failed to a non Revert error.
+    /// Returns `None` if executing the transactions failed to a non Revert error.
     /// Returns the touched+modified state of the transaction.
     ///
-    /// Note: Since here are no ordering guarantees this won't the state the tx produces when
+    /// Note: Since here are no ordering guarantees this won't the state the txs produce when
     /// executed sequentially.
-    fn transact(self, tx: Recovered<N::SignedTx>) -> Option<EvmState> {
+    fn transact_batch(
+        self,
+        txs: &[Recovered<N::SignedTx>],
+        sender: Sender<PrewarmTaskEvent>,
+    ) -> Option<()> {
         let (mut evm, evm_config, metrics) = self.evm_for_ctx()?;
 
-        // create the tx env and reset nonce
```

Comment:
out of curiosity, why are we no longer resetting the nonce? i see that in the single-tx scenario we were not doing it, only the comment remained

Reply:
I think this is a an older revm relict where we had to set the nonce to None